conan/tools/gnu/pkgconfigdeps/pc_info_loader.py

```diff
@@ -0,0 +1,139 @@
+    return pkg_aliases or []
+
+
+def _get_component_aliases(dep, comp_name):
+    if comp_name not in dep.cpp_info.components:
+        # foo::foo might be referencing the root cppinfo
+        if dep.ref.name == comp_name:
+            return _get_package_aliases(dep)
+        raise ConanException("Component '{name}::{cname}' not found in '{name}' "
+                             "package requirement".format(name=_get_package_reference_name(dep),
+                                                          cname=comp_name))
+    comp_aliases = dep.cpp_info.components[comp_name].get_property("pkg_config_aliases",
+                                                                   "PkgConfigDeps")
+    return comp_aliases or []
+
+
+def _get_package_name(dep):
+    pkg_name = dep.cpp_info.get_property("pkg_config_name", "PkgConfigDeps")
+    return pkg_name or _get_package_reference_name(dep)
+
+
+def _get_component_name(dep, comp_name):
+    if comp_name not in dep.cpp_info.components:
+        # foo::foo might be referencing the root cppinfo
+        if dep.ref.name == comp_name:
+            return _get_package_name(dep)
+        raise ConanException("Component '{name}::{cname}' not found in '{name}' "
+                             "package requirement".format(name=_get_package_reference_name(dep),
+                                                          cname=comp_name))
+    comp_name = dep.cpp_info.components[comp_name].get_property("pkg_config_name", "PkgConfigDeps")
+    return comp_name
+
+
+_PCInfoComponent = namedtuple("PCInfoComponent", ['name', 'requires', 'cpp_info', 'ref_name'])
+_PCInfoPackage = namedtuple('PCInfoPackage', ['name', 'requires'])
+
+
+class PCInfoLoader(object):
+
+    def __init__(self, conanfile, dep):
+        self._conanfile = conanfile
+        self._dep = dep
+        self._pkg_name = _get_package_name(dep)
+        self._pkg_ref_name = _get_package_reference_name(dep)
+        # Initialized variables
+        self._pkg_info = None
+        self._components_info = []
+        self._aliases_info = {}
+
+    @property
+    def package_info(self):
+        return self._pkg_info
+
+    @property
+    def components_info(self):
+        return self._components_info
+
+    @property
+    def aliases_info(self):
+        return self._aliases_info
+
+    def _get_component_requires_names(self, cpp_info):
+        """
+        Get all the pkg-config valid names from the requires ones given a dependency and
+        a CppInfo object.
+
+        Note: CppInfo could be coming from one Component object instead of the dependency
+        """
+        ret = []
+        for req in cpp_info.requires:
+            pkg_ref_name, comp_ref_name = req.split("::") if "::" in req \
+                                                          else (self._pkg_ref_name, req)
+            # FIXME: it could allow defining requires to not direct dependencies
+            req_conanfile = self._conanfile.dependencies.host[pkg_ref_name]
```

Comment:
Are you sure these are the right dependencies? Shouldn't it be the ``dep`` dependencies, instead of the root consumer ``conanfile.py``?